dallinger/experiment_server/experiment_server.py

```diff
@@ -550,6 +550,21 @@ def create_participant(worker_id, hit_id, assignment_id, mode):
     defined in reference to the participant object. You must specify the
     worker_id, hit_id, assignment_id, and mode in the url.
     """
+    fingerprint_hash = request.args.get('fingerprint_hash')
+    try:
+        fingerprint_found = models.Participant.query.\
+            filter_by(fingerprint_hash=fingerprint_hash).one_or_none()
+    except NoResultFound:
```

Comment:
Yes that's right, so after looking into it, there are two ways of doing it, .one_or_none() which does what you said, and set a catch for MultipleResultsFound, or one() and set a catch for both errors (NoResultFound and MultipleResultsFound). I'll remove the unreached catch and leave the one_or_none() since the rest of the calls seem to be using it.